### ISSUE TITLE
[Issue]: ROCdbgapi-rocm-6.1.1 fails to build with Clang-19: `no member named 'n_next' in 'doubly_linked_entry_t`

### ISSUE THREAD

**Issue by AngryLoki** [ROCm-Developer-Tools/ROCdbgapi#12]
### Problem Description

Hi, 

Clang 19 is now smart enough to find issues in template code, even if this template was never instantiated.

Issue is here: https://github.com/ROCm/ROCdbgapi/blob/rocm-6.1.1/src/utils.h#L497-L505

Build fails with:
```
/var/tmp/portage/dev-libs/rocdbgapi-6.1.1/work/ROCdbgapi-rocm-6.1.1/src/utils.h:497:13: error: no member named 'n_next' in 'doubly_linked_entry_t<T, Tag>'; did you mean 'm_next'?
  497 |     m_prev->n_next = this;
      |             ^~~~~~
      |             m_next
/var/tmp/portage/dev-libs/rocdbgapi-6.1.1/work/ROCdbgapi-rocm-6.1.1/src/utils.h:490:34: note: 'm_next' declared here
  490 |   doubly_linked_entry_t<T, Tag> *m_next{ nullptr };
      |                                  ^
/var/tmp/portage/dev-libs/rocdbgapi-6.1.1/work/ROCdbgapi-rocm-6.1.1/src/utils.h:505:13: error: no member named 'n_next' in 'doubly_linked_entry_t<T, Tag>'; did you mean 'm_next'?
  505 |     m_prev->n_next = this;
      |             ^~~~~~
      |             m_next
/var/tmp/portage/dev-libs/rocdbgapi-6.1.1/work/ROCdbgapi-rocm-6.1.1/src/utils.h:490:34: note: 'm_next' declared here
  490 |   doubly_linked_entry_t<T, Tag> *m_next{ nullptr };
      |                                  ^
2 errors generated.
```

Looks like it is just a typo and n_next should be replaced with m_next.

### Operating System

Gentoo

### CPU

-

### GPU

AMD Instinct MI300X

### ROCm Version

ROCm 6.1.0

**Comment by lancesix** [comment]
Hi, thanks for the catch and the report.

I am preparing a patch against our internal repo to remove the unused `doubly_linked_list_t` type.  I am not sure yet when it will reach github (we are working towards reducing this time).  In the meantime, do you need a patch you can use locally to fix that issue (replacing `n_next` with `m_next` should work as a minimal patch)?

**Comment by AngryLoki** [comment]
@lancesix , thank you for quick response!
No, patch is not needed, `sed -e "s/->n_next/->m_next/g" -i src/utils.h` fixed the issue. I just want to remove this temporary fix in some future version.

**Comment by lancesix** [comment]
Fixed in 0f41e80b7c0d4d0ad41bff43e07baf2fa2c1e6e9.
